### PR TITLE
[AKS] Support add Spot node pool

### PR DESCRIPTION
**Pull request by gtxistxgao**
**Description**  

Support add Spot node pool as part of GA work item.

https://docs.microsoft.com/en-us/azure/aks/spot-node-pool

**Testing Guide**  

az ask nodepool add -g resourceGroup --cluster-name clusterName -n spotPoolName --priority Spot --spot-max-price 88.88888

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 1]
```diff
@@ -256,6 +257,39 @@ def validate_taints(namespace):
...
 
 
+def validate_priority(namespace):
+    """Validates the node pool priority string."""
+    if namespace.priority is not None:
+        if namespace.priority == '':
+            return
+        if namespace.priority != "Spot" and \
+                namespace.priority != "Regular":
+            raise CLIError("--priority can only be Spot or Regular")
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py:267]: `--priority` can also be an empty string, right ?

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 2]
```diff
@@ -256,6 +257,39 @@ def validate_taints(namespace):
...
+
+
+def validate_eviction_policy(namespace):
+    """Validates the node pool priority string."""
+    if namespace.eviction_policy is not None:
+        if namespace.eviction_policy == '':
+            return
+        if namespace.eviction_policy != "Delete" and \
+                namespace.eviction_policy != "Deallocate":
+            raise CLIError("--eviction-policy can only be Delete or Deallocate")
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py:277]: Same with aforementioned comment.

**Comment by yungezz** [comment]
hi @gtxistxgao which release you'd like this one to be in? From comments, this is GA item, seems has specific timeline

**Comment by arrownj** [comment]
Hi @gtxistxgao , this is just a reminder that the CC date for next release is 9/30. Does this PR need to catch this release ?

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py (hunk 1 above)
Comment (gtxistxgao) [reply src/azure-cli/azure/cli/command_modules/acs/_validators.py:267]: when get to here. It's already not none "if namespace.priority is not None:"

**Comment by gtxistxgao** [comment]
> Hi @gtxistxgao , this is just a reminder that the CC date for next release is 9/30. Does this PR need to catch this release ?

It's not that urgent as RP's change not ready yet. Sorry for the confusion.

**Comment by gtxistxgao** [comment]
@arrownj the RP change is in. Please merge this one so we can catch up next release.

**Comment by gtxistxgao** [comment]
> @arrownj the RP change is in. Please merge this one so we can catch up next release.

@arrownj  ping... can you help merge it please.

**Review by yungezz — APPROVED** [review]

**Pull request merged**